src/main/java/jenkins/plugins/slack/SlackNotifier.java

```diff
@@ -427,8 +430,15 @@ public SlackService newSlackService(AbstractBuild r, BuildListener listener) {
         authToken = env.expand(authToken);
         authTokenCredentialId = env.expand(authTokenCredentialId);
         room = env.expand(room);
+        final String populatedToken = CredentialsObtainer.getTokenToUse(authTokenCredentialId, r.getProject(), authToken);
+        if (populatedToken != null) {
+            return new StandardSlackService(baseUrl, teamDomain, botUser, room, false, populatedToken);
+        } else {
+            logger.log(Level.INFO, "No explicit token specified and could not obtain credentials for id: " + authTokenCredentialId);
```

Comment:
this should throw an exception depending, or at least log a message to the build log as well
It shouldn't return a slack service as its not configured correctly

Reply:
Since now getTokenToUse now throws an exception (as requested in another comment) this code is now unreachable so I'm removing it.

Reply:
... however, this will cause current misconfigured jobs to crash on notification calls, is that desirable? Should I surround the calls to slackFactory (which ends up calling this) with try catch blocks?

Reply:
Yes given that the existing code does this we should keep doing it for now.
https://github.com/jenkinsci/slack-plugin/blob/b3a926151572e25ba03c8d6829d94a60a261a57c/src/main/java/jenkins/plugins/slack/StandardSlackService.java#L178